hbase-client/src/main/java/org/apache/hadoop/hbase/shaded/protobuf/ProtobufUtil.java

```diff
@@ -304,6 +305,24 @@ public static boolean isClassLoaderLoaded() {
     return classLoaderLoaded;
   }
 
+  private static final String PARSE_FROM = "parseFrom";
+
+  // We don't bother using the dynamic CLASS_LOADER above, because currently we can't support
+  // optimizing dynamically loaded classes. We can do it once we build for java9+, see the todo
+  // in ReflectedFunctionCache
+  private static final ReflectedFunctionCache<byte[], Filter> FILTERS = ReflectedFunctionCache
+    .create(ProtobufUtil.class.getClassLoader(), Filter.class, byte[].class, PARSE_FROM);
```

Comment:
Is the content of the directory specified in `hbase.dynamic.jars.dir` included in the classpath that is under the domain of this classloader? I think that if there are user-provided Filter classes in the path, we should load them. I guess that we cannot assume that they will be in the `o.a.h.h.filter` package, so we'd have to relax our class selection criteria.

Reply:
Correct we'd need to expand our package search, which may lead to increased start times. 

I've had this branch hanging around for a long time, I'd like to get this shipped and then we can tackle custom filters in a follow up when I or someone has time